rust/connlib/tunnel/src/device_channel.rs

```diff
@@ -70,7 +71,27 @@ impl Device {
         }
     }
 
-    #[cfg(any(target_os = "android", target_os = "linux"))]
+    pub(crate) fn set_tun(&mut self, tun: Tun) {
+        self.tun = Some(tun);
+
+        if let Some(waker) = self.waker.take() {
+            waker.wake();
```

Comment:
What does the waker do? Make sure that connlib updates things on its side when the Client sets a new tunnel or makes other changes?

Reply:
okay so without a `Tun` device `poll_read` could be stuck in an infinite `Poll::Pending` and this wakes it up, makes sense